server/tso/global_allocator.go

```diff
@@ -204,94 +303,95 @@ func (gta *GlobalTSOAllocator) SyncMaxTS(ctx context.Context, dcLocationMap map[
 			}
 		}
 		// Prepare to make RPC requests concurrently
-		respCh := make(chan *pdpb.SyncMaxTSResponse, len(leaderURLs))
-		errCh := make(chan error, len(leaderURLs))
-		var errList []error
+		respCh := make(chan *syncResp, len(leaderURLs))
 		wg := sync.WaitGroup{}
+		request := &pdpb.SyncMaxTSRequest{
+			Header: &pdpb.RequestHeader{
+				SenderId: gta.allocatorManager.member.ID(),
+			},
+			SkipCheck: skipCheck,
+			MaxTs:     maxTSO,
+		}
 		for _, leaderURL := range leaderURLs {
 			leaderConn, err := gta.allocatorManager.getOrCreateGRPCConn(ctx, leaderURL)
 			if err != nil {
 				return err
 			}
+			// Send SyncMaxTSRequest to all allocator leaders concurrently.
 			wg.Add(1)
-			go func(ctx context.Context, conn *grpc.ClientConn, respCh chan<- *pdpb.SyncMaxTSResponse, errCh chan<- error) {
+			go func(ctx context.Context, conn *grpc.ClientConn, respCh chan<- *syncResp) {
 				defer wg.Done()
-				request := &pdpb.SyncMaxTSRequest{
-					Header: &pdpb.RequestHeader{
-						SenderId: gta.allocatorManager.member.ID(),
-					},
-				}
-				if maxTSO.GetPhysical() != 0 {
```

Comment:
I have a question about this logic in the master. if the first time in the collecting phase, and hit the `unsyncedDCs` logics, we will retry it. but in the retry collecting phase, the maxTSO is not `Nil`, will it writes the maxTSO to local DCs? This means that instead of retrying the collection, we retry the setting max tso to local allocators.

Reply:
You are right. This is unexpected behavior. I make a PR to fix this: #3646.